### PR TITLE
Relax parallel gem version

### PR DESCRIPTION
**Pull request by mopp**
Thanks for this nice gem 🌟 
I want to use this gem in another gem and have faced trouble.

## Overview

This PR relaxes the parallel gem version constraint.

## Background

I tried to use code_keeper in a gem.
The gem already uses the parallel gem, and then the bundler selects code_keeper 0.4.0 without specifying a version due to a parallel gem version constraint.

```ruby
gem 'parallel'
gem 'code_keeper' # Add this line after bundle install (the latest parallel gem was already installed)
```

Then, I gave the code_keeper version in my Gemfile, which causes downgrading parallel gem to 1.20.1 (it was specified in the code_keeper gemspec)

```ruby
gem 'parallel'
gem "code_keeper", "~> 0.6.0"
```

```log
❯ bundle update code_keeper
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Resolving dependencies...
Using parallel 1.20.1 (was 1.24.0)
~~~~~~~
Using code_keeper 0.6.0 (was 0.4.0)
Bundle updated!
```

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,16 +7,15 @@ jobs:
...
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Ruby
       uses: ruby/setup-ruby@v1
       with:
         ruby-version: ${{ matrix.ruby }}
     - name: Run the default task
       run: |
-        gem install bundler -v 2.2.3
```
Comment (mopp) [inline .github/workflows/main.yml:20]: nit change: We don't need to specify the bundler version

> If there is a Gemfile.lock file (or $BUNDLE_GEMFILE.lock or gems.locked) with a BUNDLED WITH section, that version of Bundler will be installed and used.

https://github.com/ruby/setup-ruby?tab=readme-ov-file#bundler

File: lib/code_keeper/finder.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ def search_recursively(file_or_dir_paths)
         if FileTest.file?(edge)
           file_or_dir_paths << edge unless file_or_dir_paths.include?(edge)
         else
-          Dir.glob(("#{edge}/**/*")).each do |path|
+          Dir.glob("#{edge}/**/*").each do |path|
```
Comment (mopp) [inline lib/code_keeper/finder.rb:23]: The latest rubocop detects

```ruby
lib/code_keeper/finder.rb:23:20: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a literal.
          Dir.glob(("#{edge}/**/*")).each do |path|
```

**Comment by mopp** [comment]
@ebihara99999 I'm sorry for sending this one with many changes 🙏

**Review by ebihara99999 — APPROVED** [review]
It's easy to work with because the description is very detailed and clear. I appreciate you also update the `.github/workflows/main.yml`, updating ruby and improvement of the  actions/checkout section. Thank you very much!!

**Comment by ebihara99999** [comment]
I'm merging this though CI of Ruby 2.7 fails. I will remove the support of Ruby 2.7 and 3.0 soon.

**Comment by ebihara99999** [comment]
Released the new version just now. Thank you again! @mopp

**Comment by mopp** [comment]
@ebihara99999 Thanks for your quick work :heart:

**Pull request merged**